### PR TITLE
[EDO-198] List available trainings in skill details

### PR DESCRIPTION
**Pull request by sberlin**
Use JHipster to generate code for the new entity `training` and include it for display in associated skills. This PR is the first of a series regarding different features of managing and displaying trainings.

### CODE REVIEW

File: src/main/webapp/app/shared/skill-details/skill-details-info.component.html [hunk 1]
```diff
@@ -52,28 +52,44 @@ <h5 jhiTranslate="teamdojoApp.teams.skills.details.validation"></h5>
...
-            <p *ngIf="achievedByTeams.length === 0"
-               jhiTranslate="teamdojoApp.teams.skills.details.achievedByNone"></p>
-            <div class="clearfix"></div>
+        <div class="col-lg-6 col-md-12">
+            <h5 *ngIf="trainings.length" jhiTranslate="teamdojoApp.teams.skills.details.trainings"></h5>
+            <h5 *ngIf="!trainings.length" jhiTranslate="teamdojoApp.teams.skills.details.noTrainings"></h5>
+            <div class="available-trainings">
+                <div *ngFor="let training of trainings">
+                    <h6>
+                        <a *ngIf="training.link" [href]="training.link" target="_blank">{{training.title}}</a>
```
Comment (sberlin) [inline src/main/webapp/app/shared/skill-details/skill-details-info.component.html]: This expects links to be valid URLs, but this is not validated when creating trainings. We could introduce a pattern in the creation form. Thoughts?

**Comment by 5Heinzelmann** [comment]
At first I had compiling errors. Excecuting gradle clean and after that `./gradlew` fixed those issues.

File: src/main/webapp/app/shared/skill-details/skill-details-info.component.html (hunk 1 above)
Comment (5Heinzelmann) [reply src/main/webapp/app/shared/skill-details/skill-details-info.component.html]: Sounds good to me. Maybe similar to the `fieldValidateRules` pattern for the expiry period for skills (with a valid URL pattern of course) ?
I found `^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$` [here.](https://www.regextester.com/94502)

File: .jhipster/Training.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            ],
+            "fieldValidateRulesMaxlength": 100
+        },
+        {
+            "fieldName": "contactPerson",
+            "fieldType": "String"
+        },
+        {
+            "fieldName": "link",
+            "fieldType": "String"
```
Comment (5Heinzelmann) [inline .jhipster/Training.json]:
```suggestion
            "fieldType": "String",
            "fieldValidateRules": [
                "maxlength"
            ],
            "fieldValidateRulesMaxlength": "255"
```

Comment (5Heinzelmann) [reply .jhipster/Training.json]: Perhaps it makes sense to define a maximum length here? So far the database complains about an URL with more than 255 characters...

File: src/main/webapp/app/shared/skill-details/skill-details-info.component.html (hunk 1 above)
Comment (sberlin) [reply src/main/webapp/app/shared/skill-details/skill-details-info.component.html]: Solved in e584280. The escaping was tricky. Java and JavaScript each interpret `[` differently inside of `[]`. Java wants it escaped, JavaScript is less opinionated. The actual regex is now `^(?:http(s)?://)?[\\w.-]+(?:\\.[\\w\\.-]+)+[\\w\\-\\._~:/?#\\[\\]@!\\$&'\\(\\)\\*\\+,;=.]+$`. Mind the unescaped `/` as it is declared as a string in JSON and HTML, not a native JavaScript Regex object.

File: .jhipster/Training.json (hunk 2 above)
Comment (sberlin) [reply .jhipster/Training.json]: I also applied this to `contactPerson` in 26e8896 as this issue was raised as a bug previously.

**Pull request merged**